File: backend/src/utils/phone.ts
// normalize เบอร์โทรไทยให้เทียบกันได้ (พอร์ตจาก se-callphone/api/src/lib/phone.js)
export function normalizePhone(raw?: string | null): string {
  if (raw == null) return '';
  const s = String(raw).trim();
  const hasPlus = s.startsWith('+');
  let d = s.replace(/\D/g, '');
  if (d.startsWith('0066')) d = d.slice(4);
  if (hasPlus && d.startsWith('66')) d = d.slice(2);
  else if (!hasPlus && d.startsWith('66') && d.length === 11) d = d.slice(2);
  if ((d.length === 8 || d.length === 9) && !d.startsWith('0')) d = '0' + d;
  return d;
}

File: mobile/lib/utils/phone.dart
// normalize เบอร์โทรให้ตรงกับฝั่ง API (backend/src/utils/phone.ts)
//  - ตัดทุกอย่างที่ไม่ใช่ตัวเลข
//  - แปลง +66 / 0066 / 66 (เมื่อชัดว่าเป็นรหัสประเทศ) -> 0
String normalizePhone(String? raw) {
  if (raw == null) return '';
  final s = raw.trim();
  final hasPlus = s.startsWith('+');

  var digits = s.replaceAll(RegExp(r'\D'), ''); // เหลือเฉพาะตัวเลข

  if (digits.startsWith('0066')) {
    digits = digits.substring(4);
  }

  if (hasPlus && digits.startsWith('66')) {
    digits = digits.substring(2);
  } else if (!hasPlus && digits.startsWith('66') && digits.length == 11) {
    digits = digits.substring(2);
  }

  if ((digits.length == 8 || digits.length == 9) && !digits.startsWith('0')) {
    digits = '0$digits';
  }

  return digits;
}
